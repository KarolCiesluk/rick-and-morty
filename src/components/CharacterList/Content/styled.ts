import styled, { keyframes } from "styled-components";

import { ReactComponent as LoadingIcon } from "./loader.svg";

const rotate = keyframes`
  to {
    transform: rotate(360deg);
  }
`;

export const Loader = styled(LoadingIcon)`
  animation: ${rotate} 2s linear infinite;
  width: 86px;
  height: auto;
  margin-top: 60px;
`;



