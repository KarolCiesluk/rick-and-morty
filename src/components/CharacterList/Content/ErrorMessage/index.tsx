import { Wrapper } from "../Wrapper";
import RefreshButton from "./RefreshButton";
import { ErrorIcon, Paragraph } from "./styled";

const ErrorMessage = () => {
  return (
    <Wrapper>
      <ErrorIcon />

      <Paragraph>
        Ooops! Something went&nbsp;wrongâ€¦
      </Paragraph>

      <Paragraph>
        Please check your network connection<br />and try again
      </Paragraph>

      <RefreshButton />
    </Wrapper >
  );
};

export default ErrorMessage;