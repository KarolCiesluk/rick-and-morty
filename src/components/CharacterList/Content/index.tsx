import ErrorMessage from "./ErrorMessage";
import { Wrapper } from "./Wrapper";
import { Loader } from "./styled";

interface ContentProps {
  status: string;
  children: React.ReactNode;
}

const Content = ({ status, children }: ContentProps) => {
  if (status === "loading") {
    return (
      <Wrapper>
        <Loader />
      </Wrapper>
    );
  }
  if (status === "error") {
    return <ErrorMessage />;
  }
  return <>{children}</>;
};

export default Content;