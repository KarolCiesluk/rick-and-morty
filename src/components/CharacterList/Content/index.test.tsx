import { HashRouter } from "react-router-dom";
import renderer from "react-test-renderer";

import Content from ".";

describe("Content component testing", () => {
  it("renders ErrorMessage when status is 'error'", () => {
    const tree = renderer
      .create(
        <HashRouter>
          <Content status="error"><div>Snapshot test</div></Content>
        </HashRouter>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders Loader when status is 'loading'", () => {
    const tree = renderer
      .create(
        <HashRouter>
          <Content status="loading"><div>Snapshot test</div></Content>
        </HashRouter>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders children prop when status is 'success'", () => {
    const tree = renderer
      .create(
        <HashRouter>
          <Content status="success"><div>Snapshot test</div></Content>
        </HashRouter>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});