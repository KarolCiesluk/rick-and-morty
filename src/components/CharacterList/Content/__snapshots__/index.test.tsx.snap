// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component testing renders ErrorMessage when status is 'error' 1`] = `
<div
  className="sc-bdvvtL bKhKNO"
>
  <svg
    className="sc-dkPtRN hbRvxX"
  >
    error.svg
  </svg>
  <p
    className="sc-hKwDye jeovgI"
  >
    Ooops! Something went wrong…
  </p>
  <p
    className="sc-hKwDye jeovgI"
  >
     Please check your network connection
    <br />
    and try again
  </p>
  <button
    className="sc-gsDKAQ frLprV"
    onClick={[Function]}
  >
    Refresh
  </button>
</div>
`;

exports[`Content component testing renders Loader when status is 'loading' 1`] = `
<div
  className="sc-bdvvtL bKhKNO"
>
  <svg
    className="sc-eCImPb eFhaSB"
  >
    loader.svg
  </svg>
</div>
`;

exports[`Content component testing renders children prop when status is 'success' 1`] = `
<div>
  Snapshot test
</div>
`;
